### PR TITLE
chore:[CDS-49763]:pipeline-studio-banner

### PR DESCRIPTION
**Pull request by namanharness**
https://user-images.githubusercontent.com/110456765/211488905-1bf7f4ee-d3e0-471a-83c1-c0a9f5c1158d.mov

With Reconcile->
image

### Summary

Adding Service and Environment v2 transition banner on pipeline and template studio
#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/EndOfLifeBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    setHover(false)
+  }
+
+  React.useEffect(() => {
+    if (showBannerSession == null) {
+      setShowBanner(true)
+      setShowBannerSession('true')
+    }
+    if (showBannerSession == 'false') setShowBanner(false)
+  }, [])
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/EndOfLifeBanner.tsx:39]: This effect has missing dependencies. `showBanner` can be obtained from `showBannerSession` so this effect is not needed.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/EndOfLifeBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const onLeave = () => {
+    setHover(false)
+  }
+
+  React.useEffect(() => {
+    if (showBannerSession == null) {
+      setShowBanner(true)
+      setShowBannerSession('true')
+    }
+    if (showBannerSession == 'false') setShowBanner(false)
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/EndOfLifeBanner.tsx:38]:
```suggestion
    if (showBannerSession === 'false') setShowBanner(false)
```

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/EndOfLifeBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    <>
+      {!NG_SVC_ENV_REDESIGN && showBanner ? (
+        <Callout className={css.callout} intent="success" icon={null} onMouseEnter={onHover} onMouseLeave={onLeave}>
+          {hover ? (
+            <Text color={Color.BLACK}>
+              Harness has made a significant update to the services and environments in CD. Services and Environments
+              now have definitions that are associated with the respective object and can be managed independent of the
+              pipeline. This change is a forward looking change that won’t impact your existing pipelines. However, we
+              do plan on reducing our support on the service v1 and environments v1 and plan to deprecate by end of
+              January. We have an automated migration tool to support our users. Please contact
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/EndOfLifeBanner.tsx:50]: Add to strings

**Review by rameshx — CHANGES_REQUESTED** [review]

File: src/modules/10-common/strings/strings.en.yaml [hunk 4]
```diff
@@ -1273,3 +1273,5 @@ supportText: Support
 openInNewTab: Open in new tab
 stayOnThisPage: 'Stay on this Page'
 leaveThisPage: 'Leave this Page'
+svcEnv2Banner: 'Harness has made a significant update to the services and environments in CD. Services and Environments now have definitions that are associated with the respective object and can be managed independent of the pipeline. This change is a forward looking change that won’t impact your existing pipelines. However, we do plan on reducing our support on the service v1 and environments v1 and plan to deprecate by end of January. We have an automated migration tool to support our users. Please contact <a href= {{support}} target="_blank" rel="noreferrer"> <b>&nbsp;support.harness.io&nbsp;</b> </a> for further questions.'
+svcEnvV2Truncated: '  Harness has made a significant update to the services and environments in CD. Services and Environments now have definitions that are associated with the respective object and can be managed ...'
```
Comment (rameshx) [inline src/modules/10-common/strings/strings.en.yaml:1277]: space at the beginning
```suggestion
svcEnvV2Truncated: 'Harness has made a significant update to the services and environments in CD. Services and Environments now have definitions that are associated with the respective object and can be managed ...'
```

**Review by rameshx — DISMISSED** [review]

File: src/modules/10-common/strings/strings.en.yaml (hunk 4 above)
Comment (namanharness) [reply src/modules/10-common/strings/strings.en.yaml]: done

**Review by rameshx — DISMISSED** [review]

**Comment by thisrohangupta** [comment]
Yes this looks good @namanharness

**Review by sonaligoyal-harness — DISMISSED** [review]

**Review by sonaligoyal-harness — DISMISSED** [review]

**Review by rameshx — DISMISSED** [review]

File: src/modules/72-templates-library/components/TemplateStudio/TemplateStudioSubHeader/TemplateStudioSubHeader.tsx [hunk 5]
```diff
@@ -92,82 +93,85 @@ const TemplateStudioSubHeader: (
...
-      className={css.subHeader}
-      height={49}
-      padding={{ right: 'xlarge', left: 'xlarge' }}
-      border={{ bottom: true, color: Color.GREY_200 }}
-      background={Color.WHITE}
-    >
-      <Layout.Horizontal height={'100%'} flex={{ alignItems: 'center', justifyContent: 'space-between' }}>
-        <TemplateStudioSubHeaderLeftView onGitBranchChange={onGitBranchChange} />
-        {!templateYamlError && (
+    <Layout.Vertical>
```
Comment (samarth-sharma31) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateStudioSubHeader/TemplateStudioSubHeader.tsx:96]: Please attach a video of the behaviour when reconcile is also visible

Comment (namanharness) [reply src/modules/72-templates-library/components/TemplateStudio/TemplateStudioSubHeader/TemplateStudioSubHeader.tsx:96]: https://user-images.githubusercontent.com/110456765/211779463-ce7a9eef-d8c0-4f08-a913-b887aeeaa69d.mov

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Pull request merged**